pkg/apis/softwarecomposition/validation/valdiation_test.go

```diff
@@ -0,0 +1,124 @@
+						{
+							Identifier: "test",
+							Ports: []softwarecomposition.NetworkPort{
+								{
+									Port:     80,
+									Name:     "UDP-80",
+									Protocol: "UDP",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "invalid port name",
+			networkNeighbors: softwarecomposition.NetworkNeighbors{
+				Spec: softwarecomposition.NetworkNeighborsSpec{
+					Ingress: []softwarecomposition.NetworkNeighbor{
+						{
+							Identifier: "test",
+							Ports: []softwarecomposition.NetworkPort{
+								{
+									Port:     80,
+									Name:     "UDP",
+									Protocol: "UDP",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrors: field.ErrorList{
+				field.Invalid(field.NewPath("spec").Child("ingress").Index(0).Child("ports").Index(0).Child("name"), "UDP", "port name must be in the format {protocol}-{port}"),
+			},
+		},
+		{
+			name: "invalid port number",
+			networkNeighbors: softwarecomposition.NetworkNeighbors{
+				Spec: softwarecomposition.NetworkNeighborsSpec{
+					Ingress: []softwarecomposition.NetworkNeighbor{
+						{
+							Identifier: "test",
+							Ports: []softwarecomposition.NetworkPort{
+								{
+									Port:     1000000,
+									Name:     "UDP-1000000",
+									Protocol: "UDP",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrors: field.ErrorList{
+				field.Invalid(field.NewPath("spec").Child("ingress").Index(0).Child("ports").Index(0).Child("port"), int32(1000000), "port must be in range 0-65535"),
+			},
+		},
+		{
+			name: "invalid port number and name",
+			networkNeighbors: softwarecomposition.NetworkNeighbors{
+				Spec: softwarecomposition.NetworkNeighborsSpec{
+					Ingress: []softwarecomposition.NetworkNeighbor{
+						{
+							Identifier: "test",
+							Ports: []softwarecomposition.NetworkPort{
+								{
+									Port:     1000000,
+									Name:     "UDP-80",
+									Protocol: "UDP",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErrors: field.ErrorList{
+				field.Invalid(field.NewPath("spec").Child("ingress").Index(0).Child("ports").Index(0).Child("port"), int32(1000000), "port must be in range 0-65535"),
+				field.Invalid(field.NewPath("spec").Child("ingress").Index(0).Child("ports").Index(0).Child("name"), "UDP-80", "port name must be in the format {protocol}-{port}"),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			actualErrors := ValidateNetworkNeighbors(&test.networkNeighbors)
+			if len(actualErrors) != len(test.expectedErrors) {
+				t.Errorf("Expected %d errors, got %d", len(test.expectedErrors), len(actualErrors))
+			}
+
+			errorsFound := 0
+			for _, actualError := range actualErrors {
+
+				for _, expectedError := range test.expectedErrors {
+					if actualError.Error() == expectedError.Error() {
+						errorsFound += 1
+					}
+				}
+			}
+			assert.Equal(t, len(test.expectedErrors), errorsFound)
```

Comment:
suggestion: with our fix to the expected errors, we can simplify the assertion

```suggestion
			assert.Equal(t, test.expectedErrors, actualErrors)
```

If you need to compare the items regardless of order, try using `assert.ElementsMatch`.